File: src/app/model/items/item-lite.ts
import {ItemName} from './item-name';
import {User} from '.././user';
import {Category} from '../categories/category';

export class ItemLite extends ItemName{

    creator: User;
    imageHeadUrl: string;
    commentaire: string;

    constructor(id:number, name:string, category:Category, creator:User, imageHeadUrl:string, commentaire:string)
    {
        super(id,name,category);
        this.creator = creator;
        this.imageHeadUrl = imageHeadUrl;
        this.commentaire = commentaire;
    }
}

File: blog-angular-backend/src/main/java/fr/malkav/modele/ItemLite.java
package fr.malkav.modele;

public class ItemLite extends ItemName {

	private User creator;
	private String imageHeadUrl;
	private String commentaire;

	public ItemLite() {
	}
	
	public ItemLite(int id, String name, Category category, User creator, String imageHeadUrl, String commentaire) {
		super(id, name, category);
		this.creator = creator;
		this.imageHeadUrl = imageHeadUrl;
		this.commentaire = commentaire;
	}

	public User getCreator() {
		return creator;
	}

	public void setCreator(User creator) {
		this.creator = creator;
	}

	public String getImageHeadUrl() {
		return imageHeadUrl;
	}

	public void setImageHeadUrl(String imageHeadUrl) {
		this.imageHeadUrl = imageHeadUrl;
	}

	public String getCommentaire() {
		return commentaire;
	}

	public void setCommentaire(String commentaire) {
		this.commentaire = commentaire;
	}

}
